### PR TITLE
Avoid depending on `TRAIT_FUNCTION_GIVEN_NAME` functionality.

### PR DESCRIPTION
**Pull request by OkonSamuel**
Needs:

- [x] StatisticalTraits 3.0 (which needs https://github.com/JuliaAI/StatisticalTraits.jl/pull/20)
- [x] ScientificTypesBase 3.0 (which needs https://github.com/JuliaAI/ScientificTypesBase.jl/pull/24)
- [x] ScientificTypes 3.0 (which needs https://github.com/JuliaAI/ScientificTypes.jl/pull/174)
- [x] MLJModelInterface 1.3.4 (which needs https://github.com/JuliaAI/MLJModelInterface.jl/pull/126)

With respect to the removal of `TRAIT_FUNCTION_GIVEN_NAME`, `trait` function and `SET_CONVENTION` from [ScientificTypesBase.jl](https://github.com/JuliaAI/ScientificTypesBase.jl/pull/24). This PR

1.  Removes `TRAIT_FUNCTION_GIVEN_NAME` functionality from **src/init.jl**, because `TRAIT_FUNCTION_GIVEN_NAME` has been deleted from `ScientificTypesBase` and as it's no longer required to set `CONVENTION` at Run-time. Since MLJ explicitly uses the `DefaultConvention` defined in [ScientificTypes.jl](https://github.com/JuliaAI/ScientificTypes.jl).
2. updates the `info` method for measures at **src/measures/measures.jl**, which previously relied on the two argument definition of `info` at [StatisticalTraits.jl](https://github.com/JuliaAI/StatisticalTraits.jl/blob/dev/src/StatisticalTraits.jl#L224) which in turn relied on the eradicated `trait` function at `ScientificTypesBase`.
3. removed all import/export of `TRAIT_FUNCTION_GIVEN_NAME` and `trait` function at **src/MLJBase.jl**
4. re-organizes code at **src/resampling.jl**. To improve performance, progress meters are no longer displayed for computations taking less than 0.1 second.

All other changes are aesthetical.
cc. @ablaom

### CODE REVIEW

**Comment by ablaom** [comment]
@OkonSamuel FYI, I've changed the base for this PR to `for-0-point-19-release`.

File: src/interface/data_utils.jl [hunk 1]
```diff
@@ -19,18 +19,17 @@ MMI.int(::FI, x; args...) = CategoricalDistributions.int(x; args...)
 MMI.classes(::FI, x) = CategoricalDistributions.classes(x)
 
 # ------------------------------------------------------------------------
-# schema
-
+# `schema`
 MMI.schema(::FI, ::Val{:table}, X; kw...) = schema(X; kw...)
+MMI.schema(::FI, ::Val{:other}, X; kw...) = nothing
```
Comment (ablaom) [inline src/interface/data_utils.jl:24]: Line 24 imparts on the `schema` method used in implementations of the MLJ model interface (running in a session in which MLJBase is loaded) *different* behaviour than the `schema` method exported by MLJBase:

```julia
using MLJBase
julia> schema("abc")
ERROR: ArgumentError: Cannot inspect the internal scitypes of a non-tabular object. 
Stacktrace:
 [1] schema(X::String, #unused#::Val{:other})
   @ ScientificTypes ~/MLJ/ScientificTypes/src/schema.jl:115
 [2] schema(X::String)
   @ ScientificTypes ~/MLJ/ScientificTypes/src/schema.jl:112
 [3] top-level scope
   @ REPL[43]:1
 [4] top-level scope
   @ ~/.julia/packages/CUDA/YpW0k/src/initialization.jl:52

julia> MLJBase.MLJModelInterface.schema("abc")

julia> 
```
I suggest we remove line 24 to ensure same behaviour; `schema(X)` should throw an error if `X` is not a table.

File: src/resampling.jl [hunk 2]
```diff
@@ -940,32 +945,32 @@ function _evaluate!(func, mach, ::CPUProcesses, nfolds, verbosity)
...
     @sync begin
-        p = Progress(nfolds,
-                     dt = 0,
-                     desc = "Evaluating over $nfolds folds: ",
-                     barglyphs = BarGlyphs("[=> ]"),
-                     barlen = 25,
-                     color = :yellow)
-        channel = RemoteChannel(()->Channel{Bool}(min(1000, nfolds)), 1)
+        if !(verbosity < 1) 
+            p = Progress(
```
Comment (ablaom) [inline src/resampling.jl:949]: `if verbosity > 0` is more readable

File: src/resampling.jl [hunk 3]
```diff
@@ -914,21 +914,26 @@ evaluate(model::Measurable, args...; cache=true, kwargs...) =
...
-                 dt = 0,
-                 desc = "Evaluating over $nfolds folds: ",
-                 barglyphs = BarGlyphs("[=> ]"),
-                 barlen = 25,
-                 color = :yellow)
+function _evaluate!(func, mach, ::CPU1, nfolds, verbosity)
+    if !(verbosity < 1) 
+        p = Progress(
+            nfolds,
+            dt = 0.1,
```
Comment (ablaom) [inline src/resampling.jl:926]: Please, let's replace these "0.1" entries with a global constant defined in src/MLJBase.jl.

File: src/resampling.jl [hunk 4]
```diff
@@ -1064,10 +1076,11 @@ function evaluate!(mach::Machine, resampling, weights,
 
     # Note: `rows` and `repeats` are ignored here
 
-    resampling isa TrainTestPairs ||
+    if !(resampling isa TrainTestPairs)
         error("`resampling` must be an "*
               "`MLJ.ResamplingStrategy` or tuple of pairs "*
               "of the form `(train_rows, test_rows)`")
```
Comment (ablaom) [inline src/resampling.jl:1082]: Don't change this now (!) but FYI the practice I am now advocating for error messages is to define a constant for every exception, as in 

```julia
const ERR_BAD_RESAMPLING = ArgumentError(
    "`resampling` must be an "*
    "`MLJ.ResamplingStrategy` or tuple of pairs "*
     "of the form `(train_rows, test_rows)`"
)
...
      some_test || throw(ERR_BAD_RESAMPLING)
```

which means we can test as below, and can make modifications to error messages in just one place.

```julia
    @test_throws MLJBase.ERR_BAD_RESAMPLING do_something_bad()
```

Similarly for warnings but using `test_logs`.

File: test/interface/interface.jl [hunk 5]
```diff
@@ -14,7 +14,7 @@ include("data_utils.jl")
 
 include("model_api.jl")
 
-@test trait(rms) == :measure
+#@test trait(rms) == :measure
```
Comment (ablaom) [inline test/interface/interface.jl:17]: Yeah, you can just delete this line.

File: test/resampling.jl [hunk 6]
```diff
@@ -124,18 +123,19 @@ end
     @everywhere begin
         nfolds = 6
         nmeasures = 2
-        func(mach, k) = ((sleep(0.01*rand(rng)); fill(1:k, nmeasures)),
+        func(mach, k) = ((sleep(0.1*rand(rng)); fill(1:k, nmeasures)),
```
Comment (ablaom) [inline test/resampling.jl:126]: Let's make this a global constant defined in src/MLJBase.jl

File: test/test_utilities.jl [hunk 7]
```diff
@@ -17,28 +17,32 @@ macro testset_accelerated(name::String, var, opts::Expr, ex)
 end
 function testset_accelerated(name::String, var, ex; exclude=[])
     final_ex = quote
-        $var = CPU1()
+       local $var = CPU1()
         @testset $name $ex
     end
-    resources = Any[CPUProcesses()]
+    resources = AbstractResource[CPUProcesses()]
     @static if VERSION >= v"1.3.0-DEV.573"
```
Comment (ablaom) [inline test/test_utilities.jl:24]: No longer supporting < 1.3

File: src/init.jl [hunk 8]
```diff
@@ -8,8 +8,6 @@ function __init__()
             RemoteChannel(() -> Channel(100), myid())
     end
 
-    TRAIT_FUNCTION_GIVEN_NAME[:measure]      = is_measure
-    TRAIT_FUNCTION_GIVEN_NAME[:measure_type] = is_measure_type
```
Comment (ablaom) [inline src/init.jl:12]: May `TRAIT_FUNCTION_GIVEN_NAME`  rest in peace.

File: src/MLJBase.jl [hunk 9]
```diff
@@ -127,7 +130,7 @@ export CPU1, CPUProcesses, CPUThreads
 export Unknown, Known, Finite, Infinite,
     OrderedFactor, Multiclass, Count, Continuous, Textual,
     Binary, ColorImage, GrayImage, Image, Table
-export scitype, scitype_union, elscitype, nonmissing, trait
+export scitype, scitype_union, elscitype, nonmissing #, trait
```
Comment (ablaom) [inline src/MLJBase.jl:133]: Yeah, you can delete `trait`

**Review by ablaom — CHANGES_REQUESTED** [review]
Thanks! 

I've tested this branch locally with the upstream PR's and with a  for MLJModels and successfully loaded all 3rd party models. I've also been able to successfully regenerate the model data.

File: src/interface/data_utils.jl (hunk 1 above)
Comment (OkonSamuel) [reply src/interface/data_utils.jl:24]: Good Catch.

File: test/test_utilities.jl (hunk 7 above)
Comment (OkonSamuel) [reply test/test_utilities.jl:24]: Yeah, I didn't notice this earlier.

File: src/resampling.jl (hunk 4 above)
Comment (OkonSamuel) [reply src/resampling.jl:1082]: Okay. Noted

**Pull request merged**